src/main/java/org/checkerframework/specimin/GetTypesFullNameVisitor.java

```diff
@@ -0,0 +1,63 @@
+package org.checkerframework.specimin;
+
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.type.ClassOrInterfaceType;
+import com.github.javaparser.ast.visitor.ModifierVisitor;
+import com.github.javaparser.ast.visitor.Visitable;
+import com.github.javaparser.resolution.UnsolvedSymbolException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A visitor that traverses a Java file's AST and creates a map, associating the file name with the
+ * set of types used inside it.
+ */
+public class GetTypesFullNameVisitor extends ModifierVisitor<Void> {
+
+  /** The directory path of the Java file. */
+  private String fileDirectory = "";
+
+  /**
+   * A map that associates the file directory with the set of fully qualified names of types used
+   * within that file.
+   */
+  private Map<String, Set<String>> fileAndAssociatedTypes = new HashMap<>();
+
+  /**
+   * Get the map of files' directories and types used within those files.
+   *
+   * @return the value of fileAndAssociatedTypes
+   */
+  public Map<String, Set<String>> getFileAndAssociatedTypes() {
+    return fileAndAssociatedTypes;
```

Comment:
This should be unmodifiable, using `Collections.unmodifiableMap` (or whatever that method is called).